src/Albany_SolverFactory.cpp

```diff
@@ -165,14 +165,44 @@ SolverFactory::createModel (const Teuchos::RCP<Application>& app,
 Teuchos::RCP<Thyra::ResponseOnlyModelEvaluatorBase<ST>>
 SolverFactory::
 createSolver (const Teuchos::RCP<const Teuchos_Comm>& solverComm,
-	      const Teuchos::RCP<ModelEvaluator>&     model,
-	      const Teuchos::RCP<ModelEvaluator>&     adjointModel)
+	      const Teuchos::RCP<ModelEvaluator>&     model_tmp,
+	      const Teuchos::RCP<ModelEvaluator>&     adjointModel_tmp)
 {
+  const auto piroParams = Teuchos::sublist(m_appParams, "Piro");
+  const auto stratList =  Piro::extractStratimikosParams(piroParams);
+
+
+  int num_parameters = model_tmp->Np();
+  if (piroParams->isSublist("Analysis")) {
+    auto analysisParams = piroParams->sublist("Analysis");
+    if (analysisParams.isSublist("ROL")) {
+      auto rolParams = analysisParams.sublist("ROL");
+      if (rolParams.isParameter("Number Of Parameters"))
+        num_parameters = rolParams.get<int>("Number Of Parameters");
+    }
+  }
+
+  Teuchos::RCP<Thyra::ModelEvaluator<ST>> model, adjointModel;
+
+  if ( model_tmp->Np() > 1) {
+    std::vector<int> p_indices(num_parameters);
+
+    for(int i=0; i<num_parameters; ++i) {
+      std::ostringstream ss; ss << "Parameter Vector Index " << i;
+      p_indices[i] = i; //rolParams.get<int>(ss.str(), i);
```

Comment:
Is this correct?

Reply:
Actually, I should modify this, thanks for catching that!